platalea/rank_eval.py

```diff
@@ -13,7 +13,7 @@ def ranking(candidates, references, correct, metric=cosine, ns=(1, 5, 10)):
 
     `correct[i][j]` indicates whether for reference item i the candidate j is correct.
     """
-    distances = cdist(references, candidates)
+    distances = metric(references, candidates)
```

Comment:
Besides fixing this bug, this change will also change the current (unintended) default behavior. By changing line 11 to 
```def ranking(candidates, references, correct, metric=cdist, ns=(1, 5, 10)):```
The default behavior would stay the same for backwards compatibility.
Either way, this looks ready to be merged.

Reply:
It's better to fix the default behavior (cosine is almost always preferred to Euclidean) than to keep backwards comp, imo.